code/datums/action.dm

```diff
@@ -95,11 +95,13 @@
 /datum/action/item_action/update_button_icon()
 	button.overlays.Cut()
 	var/obj/item/I = target
+	var/old_layer = I.layer
+	var/old_plane = I.plane
 	I.layer = ABOVE_HUD_LAYER
 	I.plane = ABOVE_HUD_PLANE
 	button.overlays += I
-	I.layer = initial(I.layer)
-	I.plane = initial(I.plane)
+	I.layer = old_layer
```

Comment:
wouldnt this be fixed by making an /image instead

Reply:
Could be, though that could be more performance intensive than this.

Reply:
Yea, this is definitely cheaper.